sonar-groovy-plugin/src/main/java/org/sonar/plugins/groovy/foundation/GroovyHighlighterAndTokenizer.java

```diff
@@ -0,0 +1,237 @@
+    GroovyLexer.LITERAL_as,
+    GroovyLexer.LITERAL_assert,
+    GroovyLexer.LITERAL_boolean,
+    GroovyLexer.LITERAL_break,
+    GroovyLexer.LITERAL_byte,
+    GroovyLexer.LITERAL_case,
+    GroovyLexer.LITERAL_catch,
+    GroovyLexer.LITERAL_char,
+    GroovyLexer.LITERAL_class,
+    GroovyLexer.LITERAL_continue,
+    GroovyLexer.LITERAL_def,
+    GroovyLexer.LITERAL_default,
+    GroovyLexer.LITERAL_double,
+    GroovyLexer.LITERAL_else,
+    GroovyLexer.LITERAL_enum,
+    GroovyLexer.LITERAL_extends,
+    GroovyLexer.LITERAL_false,
+    GroovyLexer.LITERAL_finally,
+    GroovyLexer.LITERAL_float,
+    GroovyLexer.LITERAL_for,
+    GroovyLexer.LITERAL_if,
+    GroovyLexer.LITERAL_implements,
+    GroovyLexer.LITERAL_import,
+    GroovyLexer.LITERAL_in,
+    GroovyLexer.LITERAL_instanceof,
+    GroovyLexer.LITERAL_int,
+    GroovyLexer.LITERAL_interface,
+    GroovyLexer.LITERAL_long,
+    GroovyLexer.LITERAL_native,
+    GroovyLexer.LITERAL_new,
+    GroovyLexer.LITERAL_null,
+    GroovyLexer.LITERAL_package,
+    GroovyLexer.LITERAL_private,
+    GroovyLexer.LITERAL_protected,
+    GroovyLexer.LITERAL_public,
+    GroovyLexer.LITERAL_return,
+    GroovyLexer.LITERAL_short,
+    GroovyLexer.LITERAL_static,
+    GroovyLexer.LITERAL_super,
+    GroovyLexer.LITERAL_switch,
+    GroovyLexer.LITERAL_synchronized,
+    GroovyLexer.LITERAL_this,
+    GroovyLexer.LITERAL_threadsafe,
+    GroovyLexer.LITERAL_throw,
+    GroovyLexer.LITERAL_throws,
+    GroovyLexer.LITERAL_trait,
+    GroovyLexer.LITERAL_transient,
+    GroovyLexer.LITERAL_true,
+    GroovyLexer.LITERAL_try,
+    GroovyLexer.LITERAL_void,
+    GroovyLexer.LITERAL_volatile,
+    GroovyLexer.LITERAL_while
+  };
+
+  private static final int[] STRINGS = {
+     GroovyLexer.STRING_CH,
+     GroovyLexer.STRING_CONSTRUCTOR,
+     GroovyLexer.STRING_CTOR_END,
+     GroovyLexer.STRING_CTOR_MIDDLE,
+     GroovyLexer.STRING_CTOR_START,
+     GroovyLexer.STRING_LITERAL,
+     GroovyLexer.STRING_NL,
+     GroovyLexer.REGEX_MATCH,
+     GroovyLexer.REGEX_FIND,
+     GroovyLexer.REGEXP_LITERAL,
+     GroovyLexer.REGEXP_CTOR_END,
+     GroovyLexer.REGEXP_SYMBOL,
+     GroovyLexer.DOLLAR_REGEXP_LITERAL,
+     GroovyLexer.DOLLAR_REGEXP_CTOR_END,
+     GroovyLexer.DOLLAR_REGEXP_SYMBOL,
+     GroovyLexer.DOLLAR,
+     GroovyLexer.ESCAPED_DOLLAR,
+     GroovyLexer.ESCAPED_SLASH
+  };
+
+  private static final int[] CONSTANTS = {
+    GroovyLexer.DIGIT,
+    GroovyLexer.DIGITS_WITH_UNDERSCORE,
+    GroovyLexer.DIGITS_WITH_UNDERSCORE_OPT,
+    GroovyLexer.HEX_DIGIT,
+    GroovyLexer.NUM_BIG_DECIMAL,
+    GroovyLexer.NUM_BIG_INT,
+    GroovyLexer.NUM_DOUBLE,
+    GroovyLexer.NUM_FLOAT,
+    GroovyLexer.NUM_INT,
+    GroovyLexer.NUM_LONG
+  };
+
+  private static final int[] COMMENTS = {
+    GroovyLexer.ML_COMMENT,
+    GroovyLexer.SH_COMMENT,
+    GroovyLexer.SL_COMMENT
+  };
+
+  private static final Map<TypeOfText, int[]> TYPE_OF_TEXT_WITH_TOKEN_TYPE = new LinkedHashMap<>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Convert this Map to an EnumMap. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1640)